### DISCUSSION TITLE
Should we use oruga for any UI component?

### DISCUSSION THREAD

**Discussion by shashkovdanil** [kodadot/nft-gallery#8504] [General]
Oruga is now used for all UI components, even something as simple as a button. The code of these components is not the best. For example, the button component has this structure button > span > span. It is not quite clear why there are two span elements inside it.

Wouldn't it be better for us to write such simple components ourselves?

**Comment by shashkovdanil** [comment]
https://primevue.org/ looks much better than Oruga

**Reply by roiLeo** [reply]
### Unpacked Size
Oruga: 6.39 MB
PrimeVue: 41.3 MB

**Reply by shashkovdanil** [reply]
No, it doesn't work like this. It supports tree-shaking and you load only what you want. Proof is here:

<img width="1477" alt="Screenshot 2023-12-10 at 22 55 01" src="https://github.com/kodadot/nft-gallery/assets/19692986/d9e7b291-13ec-4cf7-9fb0-b5a05038ea42">

<img width="787" alt="Screenshot 2023-12-10 at 22 56 20" src="https://github.com/kodadot/nft-gallery/assets/19692986/63b765b1-bb19-422e-b49e-65b8a3dbe0fb">

I'm not an Oruga hater, but many components there are made very poorly, which can cause problems and bugs.

For example, dropdown doesn't support keyboard control, you can't move between items using arrow buttons and select an item by pressing Enter. Keyboard control is the rule of good and accessible UX.

The second problem is that ]append to body] doesn't work correctly. Any popup interface should be appended to body by default to avoid z-index bugs.

Anyway, I'm not calling anyone to anything, just sharing my thoughts and observations
